Modules/Forum/classes/class.ilObjForumGUI.php

```diff
@@ -3078,22 +3097,49 @@ public function viewThreadObject()
                     }
                 }
                 $this->renderPostContent($threadContentTemplate, $node, $this->requestAction, $pageIndex, $postIndex);
-                $this->renderDraftContent($threadContentTemplate, $this->requestAction, $render_drafts, $node, $selected_draft_id);
+                if ( ilForumPostDraft::isSavePostDraftAllowed() && $_SESSION['viewmode'] < ilForumProperties::VIEW_DATE) {
+                    $this->renderDraftContent(
+                        $threadContentTemplate,
+                        $this->requestAction,
+                        $render_drafts,
+                        $node,
+                        $selected_draft_id
+                    );
+                }
 
                 $postIndex++;
             }
 
+            if (ilForumPostDraft::isSavePostDraftAllowed()
+                && $currentSortation == ilForumProperties::VIEW_DATE_ASC
+                && $pageIndex === (int)(ceil($numberOfPostings / $pageSize) - 1)) {
+                $sortedDrafts = ilForumPostDraft::readSortedDrafts($this->user->getId(),
+                    $this->objCurrentTopic->getId());
+
+                /** @var  $sortedDraft ilForumPostDraft */
+                foreach ($sortedDrafts as $sortedDraft) {
+                    $this->renderDraftContent($threadContentTemplate,
+                        $this->requestAction,
+                        $render_drafts,
+                        $firstNodeInThread,
+                        $sortedDraft->getDraftId()
+                    );
+                }
+            }
+
             if ($firstNodeInThread instanceof \ilForumPost) {
                 if (!$this->objCurrentTopic->isClosed() && in_array($this->requestAction, ['showdraft', 'editdraft'])) {
                     $this->renderPostingForm($threadContentTemplate, $frm, $firstNodeInThread, $this->requestAction);
                 }
-                $this->renderDraftContent(
-                    $threadContentTemplate,
-                    $this->requestAction,
-                    $render_drafts,
-                    $firstNodeInThread,
-                    $selected_draft_id
-                );
+                if (ilForumPostDraft::isSavePostDraftAllowed() && $currentSortation < ilForumProperties::VIEW_DATE) {
```

Comment:
Why do you use the `<` operator here? I'd expect something like `===` or `!=='.

Reply:
No. Because the "$currentSortation === ilForumProperties::VIEW_TREE" condition does not match in every case when it is expected. 
I am still not sure why this happens, but the $currentSortation variable isn't even set properly in every changed viewmode case. 

The tests showed, that the session var is set properly instead.